modules/core/src/main/java/org/apache/ignite/internal/processors/rest/GridRestProcessor.java

```diff
@@ -145,6 +148,10 @@
     /** */
     private final Thread sesTimeoutCheckerThread;
 
+    /** Request / response listeners. */
+    private final List<BiConsumer<GridRestRequest, IgniteInternalFuture<GridRestResponse>>> lsnrs =
```

Comment:
All we need is made `handleAsync0` protected and extend `GridRestProcessor` in the plugin like this:

```
    /** */
    private static class MyGridRestProcessor extends GridRestProcessor {
        /** @param ctx Context. */
        public MyGridRestProcessor(GridKernalContext ctx) {
            super(ctx);
        }

        /** {@inheritDoc} */
        @Override protected IgniteInternalFuture<GridRestResponse> handleAsync0(GridRestRequest req) {
            IgniteInternalFuture<GridRestResponse> fut = super.handleAsync0(req);

            // Do whatever you want.

            return fut;
        }
    }
```